docs/01-app/01-getting-started/08-caching-and-revalidating.mdx

```diff
@@ -0,0 +1,398 @@
+
+export default async function BlogList() {
+  'use cache'
+  cacheLife('hours')
+
+  const posts = await getPosts()
+  return (
+    <div>
+      {posts.map((post) => (
+        <li key={post.id}>{post.title}</li>
+      ))}
+    </div>
+  )
+}
+```
+
+### Invalidating cache tags on-demand
+
+You can use the [`cacheTag` function](/docs/app/api-reference/functions/cacheTag) to define a tag for a cache entry. The entry can then be invalidated using the [`revalidateTag`](/docs/app/api-reference/functions/revalidateTag) function in a [Server Function](https://react.dev/reference/rsc/server-functions) or [Route Handler](/docs/app/building-your-application/routing/route-handlers).
+
+For example, to update the list of blog posts once a new post is created, add the `cacheTag` function to the component that renders the list of posts:
+
+```tsx filename="app/ui/posts.tsx" highlight={6} switcher
+import { cacheTag } from 'next/cache'
+
+export function Posts({ posts }) {
+  'use cache'
+  cacheTag('blog-posts')
+  return (
+    <ul>
+      {posts.map((post) => (
+        <li key={post.id}>{post.title}</li>
+      ))}
+    </ul>
+  )
+}
+```
+
+```jsx filename="app/ui/posts.js" highlight={6} switcher
+import { cacheTag } from 'next/cache'
+
+export function Posts({ posts }) {
+  'use cache'
+  cacheTag('blog-posts')
+  return (
+    <ul>
+      {posts.map((post) => (
+        <li key={post.id}>{post.title}</li>
+      ))}
+    </ul>
+  )
+}
+```
+
+Then, in the Server Action, call `revalidateTag` using the same tag to purge the cache entry:
+
+```tsx filename="app/actions.ts" switcher
+'use server'
+
+import { revalidateTag } from 'next/cache'
+
+export async function createPost() {
+  // Mutate data
+  // ...
+
+  // Purge cache
+  revalidateTag('blog-posts')
+}
+```
+
+```jsx filename="app/actions.js" switcher
+'use server'
+
+import { revalidateTag } from 'next/cache'
+
+export async function createPost() {
+  // Mutate data
+  // ...
+
+  // Purge cache
+  revalidateTag('blog-posts')
+}
+```
+
+Alternatively, you can call the [`revalidatePath`](/docs/app/api-reference/functions/revalidatePath) function to purge the whole `/blog` route (in which case you don't need `cacheTag`):
+
+```tsx filename="app/actions.ts" switcher
+'use server'
+
+import { revalidatePath } from 'next/cache'
+
+export async function createPost() {
+  // Mutate data
+  // ...
+
+  // Purge cache
+  revalidatePath('/blog')
+}
+```
+
+```tsx filename="app/actions.ts" switcher
+'use server'
+
+import { revalidatePath } from 'next/cache'
+
+export async function createPost() {
+  // Mutate data
+  // ...
+
+  // Purge cache
+  expirePath('/blog')
+}
+```
+
+## Client-side caching
+
+Next.js has a client-side cache that stores the RSC payload of route segments: [layouts](/docs/app/api-reference/file-conventions/layout), [pages](/docs/app/api-reference/file-conventions/page), and [loading states](/docs/app/api-reference/file-conventions/loading).
```

Comment:
I wonder if we want to use "RSC payload" here

Reply:
Yes, it'll be nicer to read.